pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -300,13 +306,54 @@ func (r *ReconcileCredentialsRequest) Reconcile(request reconcile.Request) (reco
 		logger.WithError(err).Error("error getting credentials request, requeuing")
 		return reconcile.Result{}, err
 	}
-	logger = logger.WithFields(log.Fields{
-		"secret": fmt.Sprintf("%s/%s", cr.Spec.SecretRef.Namespace, cr.Spec.SecretRef.Name),
-	})
+
 	// Maintain a copy, but work on a copy of the credentials request:
 	origCR := cr
 	cr = cr.DeepCopy()
 
+	// Copy deprecated secretRef to the new storage field for internal processing
+	if cr.Spec.StorageDestination.Type == "" {
+		cr.Spec.StorageDestination.Type = minterv1.SecretStorageDestinationType
+	}
+	cr.Spec.StorageDestination.Secret = &minterv1.SecretStorageDestination{
```

Comment:
always performing this operation means that the contents of the spec.SecretRef is always the canonical location of the secret data (as updating spec.storageDestination.Secret would be a no-op). is that the intended behavior?

Reply:
good call.  information in the new field should supercede information in the old field.  makes me wonder if we should fail validation and set and error condition on the CR if both are specified...